lib/rubocop/cop/rspec/preferred_shared_example_methods.rb

```diff
@@ -0,0 +1,138 @@
+      #   # good
+      #   shared_context
+      #
+      #   # good - with alias defined in Language/SharedGroups/Context
+      #   shared_scenarios
+      #
+      # @example PreferredContextMethod: shared_context
+      #   # bad - with alias defined in Language/SharedGroups/Context
+      #   shared_scenarios
+      #
+      #   # good
+      #   shared_context
+      #
+      # @example PreferredIncludeExamplesMethod: nil (default)
+      #   # good
+      #   it_behaves_like
+      #   it_should_behave_like
+      #   include_examples
+      #
+      #   # good - with alias defined in Language/Includes/Examples
+      #   include_scenarios
+      #
+      # @example PreferredIncludeExamplesMethod: it_behaves_like
+      #   # bad
+      #   shared_examples_for
+      #   shared_scenarios
+      #
+      #   # bad - with alias defined in Language/Includes/Examples
+      #   include_scenarios
+      #
+      #   # good
+      #   shared_examples
+      #
+      # @example PreferredIncludeContextMethod: nil (default)
+      #   # good
+      #   include_context
+      #
+      #   # good - with alias defined in Language/Includes/Context
+      #   include_state
+      #
+      # @example PreferredIncludeContextMethod: include_context
+      #   # bad - with alias defined in Language/Includes/Context
+      #   include_state
+      #
+      #   # good
+      #   include_context
+      #
+      class PreferredSharedExampleMethods < Base
+        extend AutoCorrector
+
+        MSG = 'Prefer `%<prefer>s` over `%<current>s`.'
+
+        # @!method shared_examples?(node)
+        def_node_matcher :shared_examples?, <<~PATTERN
+          {
+            (send #rspec? #SharedGroups.all ...)
+            (send nil? #Includes.all ...)
+          }
+        PATTERN
+
+        def on_send(node)
+          return unless shared_examples?(node)
+
+          preferred_method_name = preferred_method_name(node.method_name)
+          return unless preferred_method_name
+          return if node.method?(preferred_method_name)
+
+          selector = node.loc.selector
+          message = format(
+            MSG,
+            prefer: preferred_method_name,
+            current: node.method_name
+          )
+
+          add_offense(selector, message: message) do |corrector|
+            corrector.replace(selector, preferred_method_name)
+          end
+        end
+
+        private
+
+        def preferred_method_name(method_name)
+          if SharedGroups.examples(method_name)
+            cop_config['PreferredExamplesMethod']
+          elsif SharedGroups.context(method_name)
+            cop_config['PreferredContextMethod']
+          elsif Includes.examples(method_name)
+            cop_config['PreferredIncludeExamplesMethod']
+          elsif Includes.context(method_name)
+            # :nocov: - simplecov is not detecting the coverage of this branch!
+            cop_config['PreferredIncludeContextMethod']
+            # :nocov:
```

Comment:
This branch has coverage (line 114 of the spec), but simplecov doesn't recognize it. Is there anything else I can do?

Reply:
I think Simplecov really complains about there not being an `else` branch, and it not being covered by tests. In other places where if/elsif’s cover all cases, we have added a “noop” else branch:
```suggestion
            cop_config['PreferredIncludeContextMethod']
          else
            # :nocov:
            :noop
            # :nocov:
```

Reply:
@bquorning thanks! I need to use `false` instead of `:noop` but this works!

Reply:
(or I could use `nil` which would make more sense, but would require a disable of `Style/EmptyElse`).

Reply:
Never mind, the `else` branch can't actually be reached, `:noop` it is!